gyp/pylib/gyp/generator/make.py

```diff
@@ -1951,13 +1951,11 @@ def _InstallableTargetInstallPath(self):
     """Returns the location of the final output for an installable target."""
     # Xcode puts shared_library results into PRODUCT_DIR, and some gyp files
     # rely on this. Emulate this behavior for mac.
-
-    # XXX(TooTallNate): disabling this code since we don't want this behavior...
```

Comment:
note the TODO here specifically for node-gyp, I wonder how much this matters still? is it just a case of disallowing end-user from specifying `shared_library` in their binding.gyp? 🤷‍♂

Reply:
Change was done in https://github.com/nodejs/node-gyp/commit/a2ed0df84e294a197e3eb36b8213db6d8116028c

If we need to keep it here, can we also apply it to node? I don't know the implications...

Reply:
I don't know what to do with this. The change is in node-gyp since 2012, the commit message doesn't explain why it was done and I can't find the pull request that introduced it.

Reply:
I'll try to remove this code in nodejs/node and see if CI is happy. Maybe the gyp files that rely on this are not ours.

Reply:
Well, removing these lines breaks the Node.js build: https://ci.nodejs.org/job/node-test-commit-linux-containered/16453/nodes=ubuntu1804_sharedlibs_shared_x64/console
So we're back to the original question: how does it matter to node-gyp?
@TooTallNate sorry to bother you, but since you committed the change, maybe you remember why it was needed?

Reply:
Will this be a blocker ? or we can merge this for now and left as it is ?

Reply:
I think this is not a blocker anymore.